### PR TITLE
Final Update to Drop User Functionality on DropUser-Test

### PR DESCRIPTION
**Pull request by mehrab7**
Upon my last testing on 11/30, you are able to click on the Drop User button on Edit User and hit save to drop the user. This changes the "active" value to false in the specific UserCourse table. When looking at the roster page, the user does not appear due to the filter that filters out inactive users. The user is not deleted, but they should not and do not appear on the roster page.

### CODE REVIEW

File: BackEndFlask/controller/Routes/User_routes.py [hunk 1]
```diff
@@ -159,6 +160,24 @@ def updateUser(user_id):
...
+            createBadResponse(f"An error occured getting course_id", course_id, "users")
+            return response
+        deleteUserWorked = set_active_status_of_user_to_inactive(user_id, course_id)
+        if(type(deleteUserWorked)==type("")):
+            print(f"[User_routes /user/<int:user_id> DELETE] An error occurred unenrolling user_id: {user_id}!")
+            createBadResponse(f"An error occured unenrolling user_id", deleteUserWorked, "users")
+            return response
+        print(f"[User_routes /user/<int:user_id> DELETE] Successfully unenrolled user_id: {user_id} in course_id: {course_id}!")
+        createGoodResponse(f"Successfully unenrolled user_id: {user_id} from course_id: {course_id}!", user_schema.dumps(deleteUserWorked), 201, "userCourses")
+        # replace json with usercourse schema dump with necessary info, and replace users at the end with usercourses
```
Comment (Wrimo) [inline BackEndFlask/controller/Routes/User_routes.py:176]: I believe this comment is out of date.

File: BackEndFlask/models/schemas.py [hunk 2]
```diff
@@ -95,6 +95,7 @@ class UserCourse(db.Model):
     user_course_id = db.Column(db.Integer, primary_key=True)
     user_id = db.Column(db.Integer, ForeignKey(User.user_id), nullable=False)
     course_id = db.Column(db.Integer, ForeignKey(Course.course_id), nullable=False)
+    active = db.Column(db.Boolean, nullable=True)
```
Comment (Wrimo) [inline BackEndFlask/models/schemas.py:98]: Why allow active to be nullable? What does it mean if a user has a null active? Seems like we should be able to say all users are either active or not.

File: BackEndFlask/models/user_course.py [hunk 3]
```diff
@@ -13,23 +13,28 @@ def get_user_courses():
...
+        return UserCourse.query.filter_by(course_id=course_id, active=True).all()
     except SQLAlchemyError as e:
         error = str(e.__dict__['orig'])
         return error
 
+# Update function to only return user_courses the student is active in DUN
 def get_user_courses_by_user_id(user_id):
     try:
-        return UserCourse.query.filter_by(user_id=user_id).all()
+        # return UserCourse.query.filter_by(user_id=user_id).all()
```
Comment (Wrimo) [inline BackEndFlask/models/user_course.py:27]: No need to keep the old code in a comment. 

Additionally, there are lots of comment that seem like development notes, not documentation for future developers. I would either remove these or reword them to make the intent clear.

File: BackEndFlask/core/__init__.py [hunk 4]
```diff
@@ -9,7 +10,9 @@
         testing()
         sys.exit(1)
 app = Flask(__name__)
+CORS(app, resources={r"/*": {"origins": "*"}})
```
Comment (Wrimo) [inline BackEndFlask/core/__init__.py:13]: Question out of my own ignorance: what does CORS do and why do we need it?

**Review by Wrimo — CHANGES_REQUESTED** [review]

**Review by Wrimo — APPROVED** [review]
I just made my recommended changes to this branch, and it is ready to be merged once @malloc-nbytes and @brianlugo1 get a chance to look at it.

**Review by brianlugo1 — APPROVED** [review]
@sah0017 LGTM!

**Comment by brianlugo1** [comment]
@sah0017 This pull request has received two approved code reviews! I am going to go ahead and merge this branch into master!

**Pull request merged**